addons/product_expiry/tests/test_stock_lot.py

```diff
@@ -685,3 +685,77 @@ def test_compute_expiration_date_from_scheduled_date(self):
         delivery.action_confirm()
 
         self.assertAlmostEqual(delivery.move_line_ids[0].expiration_date, expiration_date, delta=delta)
+
+    def test_assign_lot_expiry_alert_to_default_user(self):
+        """ Test lot expiry alert is assigned to the default user of the activity type """
+
+        # User A will be the Default User on the activity type
+        user_a, user_b = self.env['res.users'].create([
+            {
+                'name': 'User A (Default)',
+                'login': 'user_a_test',
+            },
+            {
+                'name': 'User B (Responsible)',
+                'login': 'user_b_test',
+            }
+        ])
+
+        activity = self.env.ref('product_expiry.mail_activity_type_alert_date_reached')
+        activity_id = activity.id
+        activity.default_user_id = user_a.id
+
+        self.productAAA = self.ProductObj.create({
+            'name': 'Product AAA',
+            'type': 'product',
+            'tracking': 'lot',
+            'company_id': self.env.company.id,
+            'responsible_id': user_b.id
+        })
+
+        # create a new lot with with alert date in the past
+        self.lot1_productAAA = self.LotObj.create({
+            'name': 'Lot 1 ProductAAA',
+            'product_id': self.productAAA.id,
+            'alert_date': fields.Date.to_string(datetime.today() - relativedelta(days=15)),
+            'company_id': self.env.company.id,
+        })
+
+        picking_in = self.PickingObj.create({
+            'picking_type_id': self.picking_type_in,
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location,
+            'state': 'draft',
+        })
+
+        move_a = self.MoveObj.create({
+            'name': self.productAAA.name,
+            'product_id': self.productAAA.id,
+            'product_uom_qty': 33,
+            'product_uom': self.productAAA.uom_id.id,
+            'picking_id': picking_in.id,
+            'location_id': self.supplier_location,
+            'location_dest_id': self.stock_location
```

Comment:
add a coma at the end of this line so that this line history is not changed if someone adds another key in the object.